### PR TITLE
Add experimental features for Fx 122: shadowrootmode, async clipboard

### PR DESCRIPTION
**Pull request by pepelsbey**
### Description

1. Adds a section to the Experimental features page
2. Adds behind-the-flag features for the Fx 122 release notes

### Motivation

To support Firefox 122 release.

### Related issues and pull requests

- https://github.com/mdn/mdn/issues/496
- https://github.com/mdn/content/issues/31102

### CODE REVIEW

File: files/en-us/mozilla/firefox/releases/122/index.md [hunk 1]
```diff
@@ -68,6 +68,13 @@ This article provides information about the changes in Firefox 122 that affect d
...
 
+- Declarative shadow DOM: `dom.webcomponents.shadowdom.declarative.enabled`
+  - : The {{htmlelement("template")}} element now supports a `shadowrootmode` attribute which can be set to either `open` or `closed`, the same values as the `mode` option of {{domxref("Element.attachShadow()", "attachShadow()")}} method. It allows the creation of a shadow DOM subtree declaratively. ([Firefox bug 1712140](https://bugzil.la/1712140))
+- Popover API: `dom.element.popover.enabled`
+  - : Displaying popovers on top of the page content is now supported via HTML attributes or JavaScript API, including styling with the new CSS [`:popover-open`](/en-US/docs/Web/CSS/:popover-open) pseudo-class and extended support for [`::backdrop`](/en-US/docs/Web/CSS/::backdrop) pseudo-element. See more in [Popover API](/en-US/docs/Web/API/Popover_API) reference. ([Firefox bug 1823757](https://bugzil.la/1823757))
```
Comment (pepelsbey) [inline files/en-us/mozilla/firefox/releases/122/index.md:74]: @dletorey please have a look at the popover description if I got it right. Thanks!

File: files/en-us/mozilla/firefox/releases/122/index.md [hunk 2]
```diff
@@ -68,6 +68,13 @@ This article provides information about the changes in Firefox 122 that affect d
...
+- Clipboard read and write: `dom.events.asyncClipboard.clipboardItem`, `dom.events.asyncClipboard.readText` and `dom.events.asyncClipboard.writeText`
+  - : The remaining part of the async {{domxref('Clipboard API')}} is now shipped behind the flag: {{domxref('ClipboardItem')}} interface and [`read()`](/en-US/docs/Web/API/Clipboard/read), [`readText()`](/en-US/docs/Web/API/Clipboard/readText), and [`write()`](/en-US/docs/Web/API/Clipboard/write) methods. A paste context menu will appear for the user to confirm when attempting to read clipboard data not provided by the same-origin page. ([Firefox bug 1809106](https://bugzil.la/1809106))
```
Comment (pepelsbey) [inline files/en-us/mozilla/firefox/releases/122/index.md:76]: @hamishwillee please have a look at the clipboard description if I got it right. Thanks!

Comment (bsmth) [reply files/en-us/mozilla/firefox/releases/122/index.md:80]: I think Hamish is on vacation until 26th Jan, FYI.

File: files/en-us/mozilla/firefox/releases/122/index.md (hunk 1 above)
Comment (dletorey) [reply files/en-us/mozilla/firefox/releases/122/index.md:78]: This looks great to me 🎉

File: files/en-us/mozilla/firefox/releases/122/index.md [hunk 3]
```diff
@@ -78,6 +78,13 @@ This article provides information about the changes in Firefox 122 that affect d
...
 
+- Declarative shadow DOM: `dom.webcomponents.shadowdom.declarative.enabled`
+  - : The {{htmlelement("template")}} element now supports a `shadowrootmode` attribute which can be set to either `open` or `closed`, the same values as the `mode` option of {{domxref("Element.attachShadow()", "attachShadow()")}} method. It allows the creation of a shadow DOM subtree declaratively. ([Firefox bug 1712140](https://bugzil.la/1712140))
+- Popover API: `dom.element.popover.enabled`
+  - : Displaying popovers on top of the page content is now supported via HTML attributes or JavaScript API, including styling with the new CSS [`:popover-open`](/en-US/docs/Web/CSS/:popover-open) pseudo-class and extended support for [`::backdrop`](/en-US/docs/Web/CSS/::backdrop) pseudo-element. See more in [Popover API](/en-US/docs/Web/API/Popover_API) reference. ([Firefox bug 1823757](https://bugzil.la/1823757))
```
Comment (bsmth) [inline files/en-us/mozilla/firefox/releases/122/index.md:84]: Minor prose suggestion ("new" doesn't age well, usually) what do you think?

```suggestion
  - : Displaying popovers on top of page content is now supported via HTML attributes or JavaScript API, including styling with the CSS [`:popover-open`](/en-US/docs/Web/CSS/:popover-open) pseudo-class and extended support for the [`::backdrop`](/en-US/docs/Web/CSS/::backdrop) pseudo-element. See the [Popover API](/en-US/docs/Web/API/Popover_API) reference for more details. ([Firefox bug 1823757](https://bugzil.la/1823757))
```

File: files/en-us/mozilla/firefox/releases/122/index.md [hunk 4]
```diff
@@ -78,6 +78,13 @@ This article provides information about the changes in Firefox 122 that affect d
...
+- Clipboard read and write: `dom.events.asyncClipboard.clipboardItem`, `dom.events.asyncClipboard.readText` and `dom.events.asyncClipboard.writeText`
+  - : The remaining part of the async {{domxref('Clipboard API')}} is now shipped behind the flag: {{domxref('ClipboardItem')}} interface and [`read()`](/en-US/docs/Web/API/Clipboard/read), [`readText()`](/en-US/docs/Web/API/Clipboard/readText), and [`write()`](/en-US/docs/Web/API/Clipboard/write) methods. A paste context menu will appear for the user to confirm when attempting to read clipboard data not provided by the same-origin page. ([Firefox bug 1809106](https://bugzil.la/1809106))
```
Comment (bsmth) [inline files/en-us/mozilla/firefox/releases/122/index.md:86]: Is this correct?

```suggestion
  - : The async {{domxref('Clipboard API')}} is now fully supported, including [`read()`](/en-US/docs/Web/API/Clipboard/read), [`readText()`](/en-US/docs/Web/API/Clipboard/readText), and [`write()`](/en-US/docs/Web/API/Clipboard/write) methods and the {{domxref('ClipboardItem')}} interface. A paste context menu will appear for the user to confirm when reading clipboard data not provided by the same-origin page. ([Firefox bug 1809106](https://bugzil.la/1809106))
```

File: files/en-us/mozilla/firefox/experimental_features/index.md [hunk 5]
```diff
@@ -99,6 +99,46 @@ HTML password input elements ([`<input type="password">`](/en-US/docs/Web/HTML/E
   </tbody>
 </table>
 
+### Declarative shadow DOM
+
+The {{htmlelement("template")}} element now supports a `shadowrootmode` attribute which can be set to either `open` or `closed`, the same values as the `mode` option of {{domxref("Element.attachShadow()", "attachShadow()")}} method. It allows the creation of a shadow DOM subtree declaratively. (See [Firefox bug 1712140](https://bugzil.la/1712140) for more details.)
```
Comment (bsmth) [inline files/en-us/mozilla/firefox/experimental_features/index.md:104]:
```suggestion
The {{htmlelement("template")}} element now supports a `shadowrootmode` attribute which can be set to either `open` or `closed`, the same values as the `mode` option of the {{domxref("Element.attachShadow()", "attachShadow()")}} method. It allows the creation of a shadow DOM subtree declaratively. (See [Firefox bug 1712140](https://bugzil.la/1712140) for more details.)
```

File: files/en-us/mozilla/firefox/releases/122/index.md [hunk 6]
```diff
@@ -78,6 +78,18 @@ This article provides information about the changes in Firefox 122 that affect d
 
 These features are newly shipped in Firefox 122 but are disabled by default. To experiment with them, search for the appropriate preference on the `about:config` page and set it to `true`. You can find more such features on the [Experimental features](/en-US/docs/Mozilla/Firefox/Experimental_features) page.
 
+- Declarative shadow DOM: `dom.webcomponents.shadowdom.declarative.enabled`
+
+  The {{htmlelement("template")}} element now supports a `shadowrootmode` attribute which can be set to either `open` or `closed`, the same values as the `mode` option of {{domxref("Element.attachShadow()", "attachShadow()")}} method. It allows the creation of a shadow DOM subtree declaratively. ([Firefox bug 1712140](https://bugzil.la/1712140))
```
Comment (bsmth) [inline files/en-us/mozilla/firefox/releases/122/index.md:83]:
```suggestion
  The {{htmlelement("template")}} element now supports a `shadowrootmode` attribute which can be set to either `open` or `closed`, the same values as the `mode` option of the {{domxref("Element.attachShadow()", "attachShadow()")}} method. It allows the creation of a shadow DOM subtree declaratively. ([Firefox bug 1712140](https://bugzil.la/1712140))
```

**Comment by bsmth** [comment]
After https://github.com/mdn/content/pull/31612/commits/24aa0ba3888ceb2a9c8067bbd9b80382f3173da5, you may have to escape the newlines in your list like so:

```
- **Popover API** - `dom.element.popover.enabled`\
  \
  Displaying popovers on top of page content is now supported via HTML attributes or JavaScript API, including styling with the CSS [`:popover-open`](/en-US/docs/Web/CSS/:popover-open) pseudo-class and extended support for the [`::backdrop`](/en-US/docs/Web/CSS/::backdrop) pseudo-element. See the [Popover API](/en-US/docs/Web/API/Popover_API) reference for more details. ([Firefox bug 1823757](https://bugzil.la/1823757))
```

**Review by bsmth — APPROVED** [review]
I've left two really minor suggestions open for adding "the" where I think it might help. Another formatting hint if you want to use normal `<ul>`s instead of `<dl>`s.

Otherwise, looking good! Leaving my +1 🎉

**Comment by bsmth** [comment]
> I added a period instead. I don’t like two `<br>`s this escaping adds. I hope it’s all right this way :)

Gotcha I wasn't sure if you wanted an empty line inside the list, as with GFM, you can do:

```markdown
* List Item

  Thing in the list
```

And it preserves the indentation:

* List Item

  Thing in the list
  
In Yari, the whitespace is stripped, apparently, so you have to add `\` backslashes to force this style:

<details>

<summary> Screenshot:</summary>

image

</details>

On the formatting topic, what do you think about bolding the feature per list item like:

```markdown
* __Topic__ `pref?`: Description
* __Popover API__ (`dom.element.popover.enabled`): Displaying popovers on top of page content is now supported via HTML attributes or JavaScript API...
* __Popover API__ - `dom.element.popover.enabled`: Displaying popovers on top of page content is now supported via HTML attributes or JavaScript API...

```

* __Popover API__ (`dom.element.popover.enabled`): Displaying popovers on top of page content is now supported via HTML attributes or JavaScript API...
* __Popover API__ - `dom.element.popover.enabled`: Displaying popovers on top of page content is now supported via HTML attributes or JavaScript API...


>Also, I added another experimental feature for you to review (Intl.Segmenter). Sorry!

No problem, I will check it out :)

**Comment by pepelsbey** [comment]
> what do you think about bolding the feature per list item like

Good point! Applied it, thanks.

**Pull request merged**